app/models/order_detail.rb

```diff
@@ -843,10 +844,17 @@ def manual_fulfilled_at=(string)
     @manual_fulfilled_at = ValidFulfilledAtDate.parse(string)
   end
 
+  def time_data
+    @time_data ||= product.time_data_for(self)
+  end
+
   private
 
   def has_completed_reservation?
-    !product.is_a?(Instrument) || (reservation && (reservation.canceled_at || reservation.actual_end_at || reservation.reserve_end_at < Time.zone.now))
+    return true unless product.timed?
```

Comment:
This is WIP, but here's what I'm doing with the null object: https://github.com/tablexi/nucore-open/commit/9d3cf661cd91df0632dc2173af3a90c2dd67e996 

It gets rid of the `timed?` method.